lib/sparkql/token.rb

```diff
@@ -10,7 +10,7 @@ module Sparkql::Token
   DECIMAL = /^\-?[0-9]+\.[0-9]+/
   CHARACTER = /^'([^'\\]*(\\.[^'\\]*)*)'/
   DATE = /^[0-9]{4}\-[0-9]{2}\-[0-9]{2}/
-  DATETIME = /^[0-9]{4}\-[0-9]{2}\-[0-9]{2}T[0-9]{2}\:[0-9]{2}\:[0-9]{2}(\.[0-9]{6})?((\+|-)[0-9]{4})?/
```

Comment:
Notice that the `:` is optional now to allow 4 numbers in a row as well. That is covered by other tests.